library/src/test/java/com/whatsapp/stringpacks/StringPacksTest.java

```diff
@@ -105,74 +105,18 @@ public void testLanguageChangeFromChineseTaiwanToEnglish() {
     stringPacks.setUp(application);
     String zhTWString = stringPacks.getString(StringPacksTestData.STRING_ID);
     assertEquals("你好，世界", zhTWString);
-
-    // Switch language to en-US
-    configuration.locale = enLocale;
-    when(resources.getString(anyInt())).thenReturn("Test");
-    stringPacks.setUp(application);
-    String enString = stringPacks.getString(StringPacksTestData.STRING_ID);
-    assertEquals("Test", enString);
   }
 
   @Test
-  public void testLanguageChangeFromHausaNigeriaToEnglish() {
+  public void testSetUpApplicationInHausaNigeria() {
     configuration.locale = haNGLocale;
-    when(stringPacksLocaleMetaDataProvider.getPackFileIdForLocale(any())).thenReturn("ha");
+    when(stringPacksLocaleMetaDataProvider.getPackFileIdForLocale(any())).thenReturn(null);
```

Comment:
why are we returning null? isn't `ha-NG` being packed with `ha`.

In this PR we are handling 3 special cases and tests for these would be great: -
1. Region Specific locale packed in a separate `.pack` file (Your test for zh-TW should be enough)
2. Multiple locales packed together. Your test for ha-NG should be enough
3. Fallback logic. Since `ha` and `ha-NG` are packed together. I think we should write a test where we try to fetch a string for `ha-NG` but it only exists in `ha` and we fallback to `ha` and return that string.

Reply:
The `null` was something I was testing. I have changed it back to `"ha"`
I have added a test case for the fallback logic and cleaned up comments. And also updated the README file with the contents.

@ashish29dec let me know if you have more feedback.